nengo/config.py

```diff
@@ -84,6 +84,10 @@ def __repr__(self):
         return "<%s[%s]{%s}>" % (type(self).__name__,
                                  self._configures.__name__, ", ".join(params))
 
+    def __setstate__(self, state):
+        for k, v in state.items():
+            setattr(self, k, v)
```

Comment:
So yeah, this happens because `self._extraparams` doesn't exist at this point in the copy/pickle... since it does exist, `if key in self._extraparams` doesn't use `__getattr__` since it can be retrieved in `__getattribute__`.

I managed to fix that particular issue by defining `__slots__` on `ClassParams`, which I think makes sense anyway, and might speed some things up. However, when you define `__slot__` you have to implement a `__getstate__` since the default behavior is to copy `self.__dict__`...

In another branch that I haven't made a PR for, I implemented `__getstate__` and `__setstate__` for `ClassParams` because the default behavior here doesn't actually work -- it may copy `_configures` and `_extraparams`, but it doesn't modify the param instances to point to the right instance of `ClassParams` since the new `ClassParams` is a copy. I'll bring over my implementation from that other branch and add some tests, as unless I'm missing something, the current tests aren't testing all of what `ClassParams` does.